### PR TITLE
Audio

### PR DESCRIPTION
**Pull request by gferraro**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -335,25 +599,25 @@ fn main() {
...
             // This is printed out from within the spawned thread.
             assert!(result.is_ok(), "Thread must have permissions to run with realtime priority, run as root user");
 
-            info!("Connecting to frame socket {}", address);
             let mut reconnects = 0;
             let mut prev_frame_num = None;
             loop {
-                info!("Entering frame-socket loop");
                 if let Ok(_) = restart_rx.try_recv() {
+                   if let Ok(config) = DeviceConfig::load_from_fs() {
```
Comment (hardiesoft) [inline src/main.rs:618]: Why reload the config here?  If there were changes to it, wouldn't they be picked up in the `notify` watcher handler?

File: src/main.rs [hunk 2]
```diff
@@ -365,114 +629,122 @@ fn main() {
...
+
+                    match SocketStream::from_address(&address, *&config.use_wifi) {
+                        Ok(mut stream) => {
+                            info!("Connected to {}, waiting for frames from rp2040", if config.use_wifi { &"tc2-frames server"} else { &"thermal-recorder unix socket"});
+                            let mut sent_header = false;
+                            let mut ms_elapsed = 0;
+                            'send_loop: loop {
+                                // Check if we need to reset rp2040 because of a config change
+                                if let Ok(_) = restart_rx.try_recv() {
+                                    if let Ok(config) = DeviceConfig::load_from_fs() {
```
Comment (hardiesoft) [inline src/main.rs:643]: Again, I feel like this should just happen once in the `notify` handler?  Maybe I'm missing something.

File: src/main.rs [hunk 3]
```diff
@@ -304,6 +566,8 @@ fn main() {
     signal_hook::flag::register(signal_hook::consts::SIGTERM, Arc::clone(&term)).unwrap();
     signal_hook::flag::register(signal_hook::consts::SIGINT, Arc::clone(&term)).unwrap();
 
+    let mut frame_acquire = !initial_config.is_audio_device().unwrap_or_default();
+
```
Comment (hardiesoft) [inline src/main.rs:570]: If we're not acquiring frames in audio mode, should we even spin up the frame-socket thread?

File: src/main.rs [hunk 4]
```diff
@@ -551,6 +825,48 @@ fn main() {
                 }
             }
 
+            if  device_config.is_audio_device().unwrap_or_default(){
+                
+                if taking_test_recoding && (Instant::now() - last_state_read).as_millis() > 1000{
```
Comment (hardiesoft) [inline src/main.rs:830]: Doing any kind of I2C comms blocks for potentially a long time – is it safe to do it here in this loop?  Maybe it doesn't matter as it only happens when a test recording is being taken?

File: src/main.rs [hunk 5]
```diff
@@ -551,6 +825,48 @@ fn main() {
...
+                            let _ = restart_tx.send(true);
+                            taking_test_recoding = true;
+                            info!("Telling rp2040 to take test recording and restarting");
+                        } 
+                        RP2040_STATE.store(state,Ordering::Relaxed);
+                        last_state_read = Instant::now();
+                    }
+                    TAKE_TEST_AUDIO.store(false,Ordering::Relaxed);
+                }
+                if !frame_acquire{
```
Comment (hardiesoft) [inline src/main.rs:856]: Can this ever be true inside `if  device_config.is_audio_device().unwrap_or_default(){..}` ?

File: src/device_config.rs [hunk 6]
```diff
@@ -788,11 +798,19 @@ impl DeviceConfig {
         *date_time_utc >= start_time && *date_time_utc <= end_time
     }
 
+    pub fn is_audio_device(&self) -> Option<bool>{
```
Comment (hardiesoft) [inline src/device_config.rs:801]: Why does this not just return a bool?  I see you unwrapping it everywhere it's used.

File: src/main.rs [hunk 7]
```diff
@@ -784,7 +1110,12 @@ fn main() {
                                         part_count = 0;
                                         let mut file = Vec::new();
                                         file.extend_from_slice(&chunk);
-                                        save_cptv_file_to_disk(file, device_config.output_dir());
+                                        let shebang = u8_slice_as_u16_slice(&file[0..2]);
```
Comment (hardiesoft) [inline src/main.rs:1113]: This would be better just as `LittleEndian::read_u16(&file[0..2])`

**Review by hardiesoft — CHANGES_REQUESTED** [review]
A few small comments, looks mostly pretty sound.  Can you run `rustfmt` over it all though, it's quite hard to read the formatting.

File: src/main.rs (hunk 3 above)
Comment (gferraro) [reply src/main.rs:570]: at the moment its got the logic for restart rp2040, so is always needed

**Pull request merged**